tox.ini

```diff
@@ -64,6 +64,27 @@ setenv =
     {[base]setenv}
     SYNAPSE_POSTGRES = 1
 
+# A test suite for the oldest supported versions of Python libraries, to catch
+# any uses of APIs not available in them.
+[testenv:py27-old]
+skip_install=True
+deps =
+    # Old automat version for Twisted
+    Automat == 0.3.0
+
+    mock
+    lxml
+commands =
+    /usr/bin/find "{toxinidir}" -name '*.pyc' -delete
+    # Make all greater-thans equals so we test the oldest version of our direct
+    # dependencies, but make the psycopg2 version one which can compile against
```

Comment:
I still think we're going to stare at this in a month's time and try to remember the history here. Could you put

"although we support psycopg2 2.6 (because that's what's in xenial), the Travis build images use postgres 10, which is incompatible with 2.6, so we make 2.7 our minimum. Similarly, blah blah pyopenssl".